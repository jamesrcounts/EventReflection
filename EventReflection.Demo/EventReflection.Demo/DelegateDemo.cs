namespace EventReflection.Demo
{
    using System;
    using System.Collections.Generic;
    using System.Linq;
    using System.Reflection;

    using ApprovalTests;
    using ApprovalTests.Reporters;

    using Microsoft.VisualStudio.TestTools.UnitTesting;

    [TestClass]
    [UseReporter(typeof(DiffReporter))]
    public class DelegateDemo
    {
        [TestMethod]
        public void GetInvocationList()
        {
            Func<bool> truth = Domain.AlwaysTrue;
            IEnumerable<MethodInfo> methods = truth.GetInvocationList().Select(d => d.Method);
            Approvals.VerifyAll(methods, string.Empty);
        }

        [TestMethod]
        public void GetMultiInvocationList()
        {
            Func<bool> truth = Domain.AlwaysTrue;
            Func<bool> truthy = Domain.NeverFalse;
            Delegate multicast = Delegate.Combine(truth, truthy);
            IEnumerable<MethodInfo> methods = multicast.GetInvocationList().Select(d => d.Method);
            Approvals.VerifyAll(methods, string.Empty);
        }

        public class Domain
        {
            public static bool AlwaysTrue()
            {
                return true;
            }

            public static bool NeverFalse()
            {
                return true;
            }
        }
    }
}