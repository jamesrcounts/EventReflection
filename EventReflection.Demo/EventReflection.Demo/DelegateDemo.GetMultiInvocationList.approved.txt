[0] = Boolean AlwaysTrue()
[1] = Boolean NeverFalse()
